templates/todo/api/python/todo/models.py

```diff
@@ -20,11 +20,12 @@ def __init__(self, *args, **kwargs):
             credential = DefaultAzureCredential()
             keyvault_client = SecretClient(self.AZURE_KEY_VAULT_ENDPOINT, credential)
             for secret in keyvault_client.list_properties_of_secrets():
-                setattr(
-                    self,
-                    keyvault_name_as_attr(secret.name),
-                    keyvault_client.get_secret(secret.name).value,
-                )
+                if secret.name == "AZURE-COSMOS-CONNECTION-STRING":
```

Comment:
Why limiting to only cosmos connection string?

Reply:
Because when using `infra/core/database` to create `Cosmos Mongo`, only one secret named "AZURE-COSMOS-CONNECTION-STRING" is created in KeyVault. The specific code is [here](https://github.com/Menghua1/azure-dev/blob/todo-nodejs-mongo-aks/templates/common/infra/bicep/core/database/cosmos/cosmos-account.bicep#L34).

But now when using `AVM` to create `Cosmos Mongo`, multiple secrets will be created. The specific code is [here](https://github.com/Azure/bicep-registry-modules/blob/main/avm/res/document-db/database-account/main.bicep#L497-L530). 
At this time, the deployment of the API will fail and the error is as follows. It will succeed only when `secret.name == "AZURE-COSMOS-CONNECTION-STRING"` is specified.
![image](https://github.com/Azure/azure-dev/assets/111940661/d2f25cb0-49a6-44ec-a06c-657a0100a6d9)

Reply:
Does the app rely on any other secrets or just that one?  Is the app fuly working like this?

Reply:
Yes, the app only relies on that one secret `AZURE-COSMOS-CONNECTION-STRING` and the app is fully working.

In addition, we have another solution: do not specify `secret.name == "AZURE-COSMOS-CONNECTION-STRING"`, define fields with the same name as all secrets in the Settings object of the [src\api\todo\models.py](https://github.com/Azure/azure-dev/blob/main/templates/todo/api/python/todo/models.py#L14) file, as shown below, which can also solve the API deployment error.

Which option do you prefer?
![image](https://github.com/Azure/azure-dev/assets/111940661/9019cee6-a472-4565-9a2d-d6943663b5fb)

Reply:
As long as the keys/secrets aren't being written to disk or exposed then that seems ok.

Reply:
We chose to not specify the `secret.name == "AZURE-COSMOS-CONNECTION-STRING"` approach and updated the code.